training/training_loop.py

```diff
@@ -133,15 +135,22 @@ def training_loop(
     resume_time             = 0.0,      # Assumed wallclock time at the beginning. Affects reporting.
     resume_with_new_nets    = False):   # Construct new networks according to G_args and D_args before resuming training?
 
-    # Initialize dnnlib and TensorFlow.
+    # Initialize dnnlib,TensorFlow and comet.ml
     tflib.init_tf(tf_config)
     num_gpus = dnnlib.submit_config.num_gpus
+    comet_installed = True
+    if comet:
+        comet_logger = CometLogger()
+        comet_installed = comet_logger.comet_installed 
+
 
     # Load training set.
     training_set = dataset.load_dataset(data_dir=dnnlib.convert_path(data_dir), verbose=True, **dataset_args)
     grid_size, grid_reals, grid_labels = misc.setup_snapshot_image_grid(training_set, **grid_args)
     misc.save_image_grid(grid_reals, dnnlib.make_run_dir_path('reals.png'), drange=training_set.dynamic_range, grid_size=grid_size)
-
+    if comet and comet_installed:
+        comet_logger.log_image(dnnlib.make_run_dir_path('reals.png'), 'Real Samples')
```

Comment:
You can make comet_logger.log_X() be noop functions if not installed

Reply:
Can you explain this more?

Reply:
Rather than checking for comet and comet_installed, you could handle that (perhaps) inside each comet_logger.log_X() to simplify the code here.